language/evm/move-to-yul/src/functions.rs

```diff
@@ -283,19 +497,23 @@ impl<'a> FunctionGenerator<'a> {
         match bc {
             Jump(_, l) => {
                 print_loc();
-                emitln!(ctx.writer, "$block := {}", get_block(l))
+                if !skip_block {
+                    emitln!(ctx.writer, "$block := {}", get_block(l))
+                }
             }
             Branch(_, if_t, if_f, cond) => {
                 print_loc();
-                emitln!(
-                    ctx.writer,
-                    "switch {}\n\
-                     case 0  {{ $block := {} }}\n\
-                     default {{ $block := {} }}",
-                    local(cond),
-                    get_block(if_f),
-                    get_block(if_t),
-                )
+                if !skip_block {
```

Comment:
Since a branch is always at the end of a basic block, your algorithm can simply construct SCG::BasicBlock(start..end) which does not contain the branch anymore (let it 'end" just before the branch).

Reply:
to make sure that I followed, "blocks" were assigned on "Branch" and "Jump", so to do it right, I want to only skip it on "Jump"? Thanks!

Reply:
Each block either ends with a branch, jump, or return. Those instructions cannot appear anywhere else than at the end of a block.